### ISSUE TITLE
Detail textures does not work on Linux

### ISSUE THREAD

**Issue by ViBENo1** [ValveSoftware/halflife#1688]
If you playing Counter-Strike on Linux and you also use some texture pack you might noticed that `r_detailtextures` command is broken. not completely but most of the time it does nothing. if you are not familiar with this check the picture below or make a search on the net. i guess all GoldSource games affected but i'm not sure.

![r_detailtextures](https://cloud.githubusercontent.com/assets/640109/14581346/e49e1424-03ea-11e6-8150-d767388a5ea0.jpg)

**Comment by djdallmann** [comment]
Look at the ground it has texture, it's working.

**Comment by ViBENo1** [comment]
@djdallmann funny guy.

**Comment by djdallmann** [comment]
![e49e1424-03ea-11e6-8150-d767388a5ea0](https://cloud.githubusercontent.com/assets/7385812/14582700/1f5eeeee-03d2-11e6-8fb3-b987c2e97376.jpg)

If you can't see the difference then you may need new glasses.

**Comment by ViBENo1** [comment]
@djdallmann jesus, it's just for demonstration! oh my god....

**Comment by djdallmann** [comment]
it's working though... one looks like sandpaper and one is smooth.

**Comment by ViBENo1** [comment]
yes...just wanted to show those who never really noticed or used this. and as you can see the pictures also came from Windows, cause hard to illustrate it on Linux since the function is not working.

**Comment by djdallmann** [comment]
sorry, it wasn't clear by your first post thanks for clarifying.

**Comment by DoMiNeLa10** [comment]
Is anyone working on this? I got the valve complete pack, and I feel like I got inferior versions of the games with the broken renderer, and no option to run them in software mode (the mesa software renderer is irrelevant, as it provides visuals on par with any OpenGL GPU). This reminds me of broken OpenGL renderers shipped with GOG releases of UE1 games which don't support detail textures as well. Games end up looking significantly worse than they looked like back in the day. The fact that D3D was removed from windows releases sucks as well, as it means I can't just use wine to get a fully functional version of the game. It makes me want to get older versions of these games via other means just to get the full experience instead of a broken hackjob I have right now.

The sad part is that it's been over 2 years since the last comment here, and nobody at Valve responded. What a great way to ruin games by releasing updates which significantly degrade them.

**Comment by ViBE-HU** [comment]
> 
> 
> Is anyone working on this? I got the valve complete pack, and I feel like I got inferior versions of the games with the broken renderer, and no option to run them in software mode (the mesa software renderer is irrelevant, as it provides visuals on par with any OpenGL GPU). This reminds me of broken OpenGL renderers shipped with GOG releases of UE1 games which don't support detail textures as well. Games end up looking significantly worse than they looked like back in the day. The fact that D3D was removed from windows releases sucks as well, as it means I can't just use wine to get a fully functional version of the game. It makes me want to get older versions of these games via other means just to get the full experience instead of a broken hackjob I have right now.
> 
> The sad part is that it's been over 2 years since the last comment here, and nobody at Valve responded. What a great way to ruin games by releasing updates which significantly degrade them.

I guess not. GoldSrc is out of priority as you see. They did some bugfixes but it's still far from perfect. But they just not interested sadly. Spamming humanity with CS:GO is much more important. Btw may I not clearly understand you but detailed textures is only working with OpenGL. This feature will not working with D3D.

**Comment by DoMiNeLa10** [comment]
@ViBE-HU the whole thing with mesa was related to the fact that I checked a couple of games with the mesa software renderer to see whether any of my problems are related to the drivers I'm using (nouveau), and the fact that the original software renderer is not avaiable in goldsrc games. Detail textures used to work in D3D, I can clearly recall playing around with them in a copy of the game I got from unofficial sources back in the day, so I'm certain they used to work properly.

**Comment by ViBE-HU** [comment]
@DoMiNeLa10 That's pretty strange since it never ever let me to use it with D3D on Windows. Just check a 4554 build. The game always hide the option with D3D. That was the reason back in the days why I used OpenGL cause it was available only for it. I'm not a dev but AFAIK it's an OpenGL related feature.

**Comment by DoMiNeLa10** [comment]
@ViBE-HU D3D was dropped entirely around 2013, older versions of goldsrc games used to have that renderer available.

**Comment by ZeroYamaoka** [comment]
In my case Detail texutes works but something is wrong with it 

![Image](https://github.com/user-attachments/assets/a78f67c7-a280-47e0-979d-7e9209523bb6)
